ui/src/ScholarReader.tsx

```diff
@@ -412,28 +413,31 @@ class ScholarReader extends React.PureComponent<ScholarReaderProps, State> {
   }
 
   scrollSymbolIntoView() {
-    setTimeout(() => {
-      const { selectedSymbol, pdfViewer, pages } = this.state;
-      if (pdfViewer && selectedSymbol) {
-        const symBounds = selectedSymbol.bounding_boxes[0];
-        const { left } = selectors.divDimensionStyles(
-          pages[symBounds.page + 1].view, symBounds
-        );
-        
-        // Left = left position on the pdf page, scroll left = left offset of the pdf
-        // 200 = extra padding from the pdf side panel. 
-        const relativeSymbolLeftPosition = left - pdfViewer.container.scrollLeft + 200;
-        // window.innerWidth = possible visible area of the viewport, 
-        // 470 = width now obscured by the open drawer. 
-        const viewableViewportWidth = window.innerWidth - 470;
-        console.log(`Left Position: ${relativeSymbolLeftPosition} > ${viewableViewportWidth} | moving: ${relativeSymbolLeftPosition - viewableViewportWidth}`);
-        if (relativeSymbolLeftPosition > viewableViewportWidth) {
-          // Here we are moving the pdf to the left by the amount the symbol was obscured by
-          // + some padding to make it a little bit away from the drawer. 
-          pdfViewer.container.scrollLeft += Math.max((relativeSymbolLeftPosition - viewableViewportWidth) + 50, 0);
-        }
-      } 
-    }, 0);
+    const { selectedSymbol, pdfViewer, pages } = this.state;
+    if (pdfViewer && selectedSymbol) {
+      const symBounds = selectedSymbol.bounding_boxes[0];
+      const { left } = selectors.divDimensionStyles(
+        pages[symBounds.page + 1].view, symBounds
+      );
+      /*
+      * Each component of the calculation: 
+      * Left = left position on the pdf page of the selected symbol
+      * scrollLeft = how much the pdf has been scrolled left already
+      * 200 = the extra padding from the pdf's side panel
+      * ----------------
+      * innerWidth = possible visible area of the viewport
+      * 470 = width of the drawer that is now obscuring the view
+      */
+      /*
+      * TODO(@zkirby) need to account for if the side panel is not open.
+      */
```

Comment:
@zkirby is this something that can make it into this PR? If not, we can do another one, just open a new issue for this. Wondering if it would be helpful to discuss alternatives for implementation here?

Reply:
I can try to look a little deeper in the PDF code and see if there is an easy way to tell without massive changes.

Reply:
@zkirby take a look at [these events](https://github.com/mozilla/pdf.js/blob/52749d1f0dd04d344310c5ac599d30e0e0a4b390/web/app.js#L1601) that get fired in pdf.js, do you think any of them are events we can track to detect the state of the sidebar?

We already listen for some events from pdf.js [here](https://github.com/allenai/scholar-reader/blob/3187c2f51f6363eac89cb07fe659f08b93d94346/ui/src/ScholarReader.tsx#L316), so could do the same for other sidebar-related events.

Reply:
Looks like that worked - thanks!!